ambry-clustermap/src/main/java/com/github/ambry/clustermap/MySqlReportAggregatorTask.java

```diff
@@ -0,0 +1,134 @@
+ */
+package com.github.ambry.clustermap;
+
+import com.github.ambry.commons.Callback;
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.server.AccountStatsStore;
+import com.github.ambry.server.StatsReportType;
+import com.github.ambry.server.StatsSnapshot;
+import com.github.ambry.server.StatsWrapper;
+import com.github.ambry.utils.Pair;
+import com.github.ambry.utils.SystemTime;
+import com.github.ambry.utils.Time;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.helix.HelixManager;
+import org.apache.helix.task.Task;
+import org.apache.helix.task.TaskResult;
+import org.apache.helix.task.UserContentStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * MySql task to aggregate container storage usage across all storage nodes and update the aggregated stats.
+ */
+public class MySqlReportAggregatorTask extends UserContentStore implements Task {
+  public static final String TASK_COMMAND_PREFIX = "mysql_aggregate";
+  private static final Logger logger = LoggerFactory.getLogger(MySqlReportAggregatorTask.class);
+  private static final ZoneOffset zoneOffset = ZoneId.systemDefault().getRules().getOffset(LocalDateTime.now());
+  static final DateTimeFormatter TIMESTAMP_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM");
+  private final HelixClusterAggregator clusterAggregator;
+  private final HelixManager manager;
+  private final StatsReportType statsReportType;
+  private final Callback<StatsSnapshot> callback;
+  private final ClusterMapConfig clusterMapConfig;
+  private final AccountStatsStore accountStatsStore;
+  private final Time time;
+
+  /**
+   * Instantiates {@link MySqlReportAggregatorTask}.
+   * @param manager The {@link HelixManager} to retrieve all storage nodes.
+   * @param relevantTimePeriodInMs relevant time period in ms within which values are considered to be valid. Values
+   *                               outside of this period will be ignored.
+   * @param statsReportType the type of stats report
+   * @param accountStatsStore The {@link AccountStatsStore} to retrieve stats and store aggregated stats.
+   * @param callback a callback which will be invoked when the aggregation report has been generated successfully.
+   * @param clusterMapConfig the {@link ClusterMapConfig} associated with helix participant.
+   */
+  MySqlReportAggregatorTask(HelixManager manager, long relevantTimePeriodInMs, StatsReportType statsReportType,
+      AccountStatsStore accountStatsStore, Callback<StatsSnapshot> callback, ClusterMapConfig clusterMapConfig) {
+    this(manager, relevantTimePeriodInMs, statsReportType, accountStatsStore, callback, clusterMapConfig,
+        SystemTime.getInstance());
+  }
+
+  MySqlReportAggregatorTask(HelixManager manager, long relevantTimePeriodInMs, StatsReportType statsReportType,
```

Comment:
Why we need to constructors here?

Reply:
This constructor is for testing, but i realize the tests are redundant, removed this constructor.